src/DatePicker/CalendarMonth.js

```diff
@@ -34,25 +31,27 @@ const CalendarMonth = React.createClass({
 
   _getDayElements(week, i) {
     return week.map((day, j) => {
-      const isSameDate = DateTime.isEqualDate(this.props.selectedDate, day);
-      const disabled = this._shouldDisableDate(day);
-      const selected = !disabled && isSameDate;
+      if (this.props.getDayElement) {
+        const component = this.props.getDayElement(day, this.props.selectedDate);
 
-      if (isSameDate) {
-        if (disabled) {
-          this._selectedDateDisabled = true;
-        } else {
-          this._selectedDateDisabled = false;
+        if (!ReactTestUtils.isElementOfType(component, DayButton)) {
+          throw 'getDayElement must return an element of type DayButton';
         }
+
+        return React.cloneElement(component, {
```

Comment:
Having a clone is going to decrease the performances.
Do we need `getDayElement()` to return a `DayButton`? Can't this be just properties to apply to the component?